runtime/interpreter/interpreter.go

```diff
@@ -3292,22 +3288,24 @@ func (interpreter *Interpreter) IsSubTypeOfSemaType(subType StaticType, superTyp
 }
 
 func (interpreter *Interpreter) domainPaths(address common.Address, domain common.PathDomain) []Value {
-	config := interpreter.SharedState.Config
-	storageMap := config.Storage.GetStorageMap(address, domain.Identifier(), false)
+	storageMap := interpreter.Storage().GetStorageMap(address, domain.Identifier(), false)
 	if storageMap == nil {
 		return []Value{}
 	}
 	iterator := storageMap.Iterator(interpreter)
-	var values []Value
+	var paths []Value
 
 	count := storageMap.Count()
 	if count > 0 {
-		values = make([]Value, 0, count)
-		for key := iterator.NextKey(); key != ""; key = iterator.NextKey() {
-			values = append(values, NewPathValue(interpreter, domain, key))
+		paths = make([]Value, 0, count)
+		for key := iterator.NextKey(); key != nil; key = iterator.NextKey() {
+			// TODO: unfortunately, the iterator only returns an atree.Value, not a StorageMapKey
```

Comment:
Is there anything to do here in the future, that is worth creating an issue? or is it already handled in the follow-up PRs?

Reply:
It would be nice to figure out a way to implement `StorageMapKey` more elegantly. The current approach works, it's just that it isn't elegant – e.g. here we have to cast. 

I left this TODO just as a general note that it would be nice to figure out a way to make it more elegant, it isn't addressed in a follow-up PR, nor does it warrant an issue.